code/modules/surgery/organs/external/wings.dm

```diff
@@ -54,6 +54,9 @@
 
 	fly.Remove(organ_owner)
 
+	if(HAS_TRAIT_FROM(organ_owner, TRAIT_MOVE_FLYING, SPECIES_FLIGHT_TRAIT))
```

Comment:
This trait isn't limited to this, wouldn't it be better to check ``wings_open``?
Would also fix edge cases of multiple wings, I guess?